File: 프로그래머스/3/12907. 거스름돈/거스름돈.py
def solution(n, money):
    MOD = 1_000_000_007
    dp = [0] * (n + 1)
    dp[0] = 1
    
    for coin in money:
        for amount in range(coin, n + 1):
            dp[amount] = (dp[amount] + dp[amount - coin]) % MOD
    
    return dp[n]

File: 프로그래머스/3/12907. 거스름돈/거스름돈.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> money) {
    const int MOD = 1000000007;
    
    vector<int> dp(n + 1, 0);
    dp[0] = 1;
    
    for (int coin : money)
    {
        for (int amount = coin; amount <= n; amount++)
        {
            dp[amount] += dp[amount - coin];
            if (dp[amount] >= MOD) dp[amount] -= MOD;
        }
    }
    
    return dp[n];
}